equinox/_module.py

```diff
@@ -477,10 +477,33 @@ def __init__(self, *args, **kwargs):
                     if not has_abstract_name:
                         # Invariant: abstract classes have names beginning with
                         # `Abstract`.
+                        inner = " "
+                        if len(cls.__abstractmethods__) > 0:
+                            inner = (
+                                inner
+                                + f"abstract methods: {list(cls.__abstractmethods__)}"
+                            )
+                        if len(cls.__abstractvars__) > 0:
+                            if len(inner) > 2:
+                                inner = inner + ", "
+                            inner = (
+                                inner
+                                + f"abstract variables: {list(cls.__abstractvars__)}"
+                            )
+                        if len(cls.__abstractclassvars__) > 0:
+                            if len(inner) > 2:
+                                inner = inner + ", "
+                            inner = (
+                                inner
+                                + f"abstract class variables: \
+                                {list(cls.__abstractclassvars__)}"
+                            )
+                        inner = inner + "."
```

Comment:
I think this woudl be simply done via something like
```python
pieces = []
if ...:
    pieces.append(some string)
...
message = ", ".join(pieces)
```
?

Reply:
Yea, they are equivalent, I changed it to that way. Neither seemed particularly more readable to me.

Reply:
Technically the repeated concatenation is quadratic ;)